src/main/java/com/minecolonies/coremod/entity/ai/util/ChatSpamFilter.java

```diff
@@ -79,6 +80,13 @@ public void requestTextComponentWithoutSpam(@NotNull final TextComponentBase cha
      */
     public void talkWithoutSpam(final String key, final Object... chat)
     {
+        if (!citizenData.getCitizenEntity().isPresent())
+        {
+            return;
+        }
+
+        final EntityCitizen worker = citizenData.getCitizenEntity().get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)